Code/Framework/AzCore/AzCore/IO/FileIO.cpp

```diff
@@ -246,6 +246,10 @@ namespace AZ::IO
             }
             if (AnyFlag(mode & OpenMode::ModeBinary))
             {
+                if (AnyFlag(mode & OpenMode::ModeRead))
+                {
+                    return "wb+";
+                }
```

Comment:
The combination of `OpenMode::ModeWrite | OpenMode::ModeUpdate | OpenMode::ModeBinary` opens the file in update mode while supporting both input and output.
These flags map to [fopen](https://pubs.opengroup.org/onlinepubs/007904975/functions/fopen.html) `mode` parameter, where the write mode(`w`) cannot be combined with the read mode(`r`) flag.